test/e2e/monitoring/prometheus_utils.go

```diff
@@ -0,0 +1,109 @@
+package test
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os/exec"
+	"strings"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	promApi "github.com/prometheus/client_golang/api"
+	promApiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	promConfig "github.com/prometheus/common/config"
+
+	testenv "github.com/kubevirt/cluster-network-addons-operator/test/env"
+
+)
+
+var portForwardCmd *exec.Cmd
+
+type promClient struct {
+	client     promApiv1.API
+	sourcePort int
+	namespace  string
+}
+
+func newPromClient(sourcePort int, monitoringNs string) *promClient {
+	prometheusClient := &promClient{
+		sourcePort: sourcePort,
+		namespace:  monitoringNs,
+	}
+
+	prometheusClient.client = initializePromClient(prometheusClient.getPrometheusUrl(), prometheusClient.getAuthorizationTokenForPrometheus())
+
+	return prometheusClient
+}
+
+func (p *promClient) checkForAlert(alertName string) {
+	Eventually(func() *promApiv1.Alert {
+		alert := p.getAlertByName(alertName)
+		return alert
+	}, 2*time.Minute, 1*time.Second).ShouldNot(BeNil(), fmt.Sprintf("alert %s not fired", alertName))
+}
+
+func (p *promClient) getPrometheusUrl() string {
+	return fmt.Sprintf("http://localhost:%d", p.sourcePort)
```

Comment:
this wont work
we are using https, please use Bearer
see please 
https://github.com/kubevirt/cluster-network-addons-operator/pull/1144/files#diff-27325d74042ddc4a9b74aa4001b272ec23f03d9cd8c52a629e04bd93eb5f9490R710

Reply:
we do use a brearer, but you are right we need to use https

Reply:
after checking, it looks like http is the only working way. not sure why though..

Reply:
from the CNAO http will work because it still runs there, kube-rbac-proxy reads it
for https, you need the bearer and to change the link as in the above comments,

with CRC it works, for k8s vanilla Erkan said it doesn't work maybe because missing RBACs
because the namespace that Prometheus adds rbac to is `openshift-monitoring`, and we have namespace `monitoring` on U/S

Reply:
>from the CNAO http will work because it still runs there, kube-rbac-proxy reads it
for https, you need the bearer and to change the link as in the above comments,
the https works in that link,

We are not accessing CNAO here like former tests, we are directly querying prometheus API. The reason we don't need https is because we are wrapping the test with a [port-forward command](https://github.com/kubevirt/cluster-network-addons-operator/pull/1250/commits/85432af6588dcefac5eefd9f0f8b8d0dce9d3cb7#diff-2915e6ecb9d8edd039184f1edf98c641cabb97148aafac11e59281f37c2b1ab3R23) to the monitoring namespace

>with CRC it works, for k8s vanilla Erkan said it doesn't work maybe because missing RBACs
because the namespace that Prometheus adds rbac to is openshift-monitoring, and we have namespace monitoring on U/S

The reason it worked on CRC is unrelated (the issue was a label selector in the prometheus CRD, taken cared of [here](https://github.com/kubevirt/cluster-network-addons-operator/pull/1250/commits/85432af6588dcefac5eefd9f0f8b8d0dce9d3cb7#diff-6c6dc8ec613a709e2ccbaed045bac1a89bd91f5d60e8f0c303b98671931cf768R33)).

Reply:
> We are not accessing CNAO here like former tests, we are directly querying prometheus API. The reason we don't need https is because we are wrapping the test with a [port-forward command](https://github.com/kubevirt/cluster-network-addons-operator/pull/1250/commits/85432af6588dcefac5eefd9f0f8b8d0dce9d3cb7#diff-2915e6ecb9d8edd039184f1edf98c641cabb97148aafac11e59281f37c2b1ab3R23) to the monitoring namespace

In case it port forward the internal http port, then it doesn't fully test e2e imho, better to consider trying to improve it if its the case, on a follow PR please

Reply:
sure. I don't see why not do it on a separate PR